parlai/tasks/coco_caption/agents.py

```diff
@@ -258,13 +260,13 @@ def next_example(self):
         ready = None
         # pull up the currently queued example
         if self.example is not None:
-            if self.image_mode != 'none':
+            if self.image_mode != 'none' and 'image_id' in self.example:
                 # move the image we loaded in the background into the example
                 image = self.data_queue.get()
                 self.example['image'] = image
-            ready = (self.example, self.epochDone)
+            ready = (self.example, self.imageEpochDone)
         # get the next base example: super().next_example() calls self.get()
-        self.example, self.epochDone = super().next_example()
+        self.example, self.imageEpochDone = super().next_example()
```

Comment:
do you use this anywhere?

Reply:
Yes. We use `imageEpochDone` instead of `epochDone` as FixedDialogTeacher takes the return of this `next_example` function and sets `self.epochDone`. So before this change, the loader livelocked as `self.epochDone` was being set to True in this line, but the prior result returned by `ready` contained `self.epochDone = False`. Therefore, FixedDialogTeacher would set `self.epochDone` to `False` immediately after this function exited. The program would see that the epoch wasn't completed and call this function again, and `ready` would once again return `False`, causing livelock even though `super().next_example()` kept returning True. Simply changing this result to `_` would mean that the trigger for `epochDone` would never be sent through to the system.

Reply:
ah you're right! rescinded